src/wp-includes/formatting.php

```diff
@@ -4041,16 +4041,76 @@ function wp_trim_excerpt( $text = '', $post = null ) {
  * @param string $text      Text to trim.
  * @param int    $num_words Number of words. Default 55.
  * @param string $more      Optional. What to append if $text needs to be trimmed. Default '&hellip;'.
+ * @param bool   $allow_tags Optional. Whether to allow HTML tags in the text. Default false.
  * @return string Trimmed text.
  */
-function wp_trim_words( $text, $num_words = 55, $more = null ) {
+function wp_trim_words( $text, $num_words = 55, $more = null, $allow_tags = false ) {
 	if ( null === $more ) {
 		$more = __( '&hellip;' );
 	}
 
+	$allowed_tags = array(
+		'a'      => array(
+			'class' => array(),
+			'style' => array(),
+			'href'  => array(),
+			'id'    => array(),
+		),
+		'strong' => array(),
+		'b'      => array(),
+		'em'     => array(),
+		'img'    => array(
+			'class'  => array(),
+			'style'  => array(),
+			'src'    => array(),
+			'alt'    => array(),
+			'width'  => array(),
+			'height' => array(),
+		),
+		'mark'   => array(
+			'class' => array(),
+			'style' => array(),
+		),
+		'code'   => array(
+			'class' => array(),
+			'style' => array(),
+		),
+		'kbd'    => array(
+			'class' => array(),
+			'style' => array(),
+		),
+		'bdo'    => array(
+			'class' => array(),
+			'style' => array(),
+			'dir'   => array(),
+			'lang'  => array(),
+
+		),
+		'span'   => array(
+			'class' => array(),
+			'style' => array(),
+		),
+		'sup'    => array(
+			'class' => array(),
+			'style' => array(),
+		),
+		'sub'    => array(
+			'class' => array(),
+			'style' => array(),
+		),
+		's'      => array(
+			'class' => array(),
+			'style' => array(),
+		),
+	);
+
 	$original_text = $text;
-	$text          = wp_strip_all_tags( $text );
-	$num_words     = (int) $num_words;
+	if ( $allow_tags ) {
```

Comment:
Can we move the array in condition?

Reply:
@mukeshpanchal27 moved the array inside the condition, please check.